dashboard/app/models/script.rb

```diff
@@ -515,16 +513,21 @@ def self.get_without_cache(id_or_name, with_associated_models: true)
   #   get_from_cache('frozen') --> script_cache['frozen'] = <Script name="frozen", id=...>
   #
   # @param id_or_name [String|Integer] script id, script name, or script family name.
-  def self.get_from_cache(id_or_name)
-    if CourseOffering.get_from_cache(id_or_name).present?
-      raise "Do not call Script.get_from_cache with a family_name. Call Script.get_unit_family_redirect_for_user instead.  Family: #{id_or_name}"
-    end
-
-    return get_without_cache(id_or_name, with_associated_models: false) unless should_cache?
-    cache_key = id_or_name.to_s
-    script_cache.fetch(cache_key) do
-      # Populate cache on miss.
-      script_cache[cache_key] = get_without_cache(id_or_name)
+  def self.get_from_cache(id_or_name, raise_exceptions: true)
+    script =
+      if should_cache?
+        cache_key = id_or_name.to_s
+        script_cache.fetch(cache_key) do
+          # Populate cache on miss.
+          script_cache[cache_key] = get_without_cache(id_or_name)
+        end
+      else
+        get_without_cache(id_or_name, with_associated_models: false)
+      end
+    return script if script
+    if raise_exceptions
+      raise "Do not call Script.get_from_cache with a family_name. Call Script.get_unit_family_redirect_for_user instead.  Family: #{id_or_name}" if Script.family_names.include?(id_or_name)
+      raise ActiveRecord::RecordNotFound.new("Couldn't find Script with id|name=#{id_or_name}")
```

Comment:
apologies, I had drafted an incomplete thought here and clicked comment by accident. this makes this more complicated, but I don't see an obvious simpler way to handle this.

Reply:
Yeah, ideally I wouldn't pass this parameter in but this method is called in a lot of places and I didn't want to risk breaking one of those places, at least in this PR.